cli/src/utils/context/mod.rs

```diff
@@ -0,0 +1,57 @@
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+use optional_struct::*;
+
+use crate::utils::constants::CONFIG_FILE;
+use crate::utils::repo::helpers::to_relative_path_to_project_root;
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("failed to open configuration file")]
+    FailedToOpenConfFile,
+    #[error("invalid configuration file")]
+    InvalidConfFile,
+    #[error("error with actors_bin_dir")]
+    FailedToGetActorsBinDirAsStr,
+}
+
+/// Context structure helping accessing the repository area in a consistent way throughout the CLI
+/// commands.
+#[optional_struct]
+#[derive(PartialEq, Debug, Serialize, Deserialize, Clone)]
+pub(crate) struct CliContext {
+    pub(crate) actors_bin_dir: PathBuf,
+}
+
+impl CliContext {
+    /// Public function helping to initialize a [ Context ] object.
+    pub(crate) fn new() -> anyhow::Result<Self> {
+        let root_path = env::current_dir()?;
+        let config_file_path = root_path.join(CONFIG_FILE);
+        // fetch config from configuration file if it exists
+        let config = if config_file_path.exists() {
+            let config_file = File::open(&config_file_path).context(Error::FailedToOpenConfFile)?;
+            serde_yaml::from_reader(config_file).context(Error::InvalidConfFile)?
+        } else {
+            OptionalCliContext::default()
```

Comment:
The idea here is that we need two kind of objects, that are related

1️⃣ an object that gathers all contextual information, which mostly holds no undefined values, but rather user-defines ones, or default ones. It is useful for instance here to know what is the _actors' bin dir_, no matter if it's been defined by the user, or if the value felt back on the default, but to be sure that there is a non-null value there.

2️⃣ an object that helps gather user-defined values. Not 100% of available contextual values may be defined by the user, who may choose to define only 1 or 2 for instance. And in terms of implementation, it can prove handy to only rely on `serde` to get values from a yaml file.

So in the end, this architecture with one config file with mostly not optionable fields, and a symmetrical _optionized_ structure to help with (de)serialization of manually defined values, seems to work fine.